csharp/tools/MauiModelTester/OrtInferenceSession.cs

```diff
@@ -0,0 +1,117 @@
+    }
+
+    // An inference session runs an ONNX model
+    internal class OrtInferenceSession
+    {
+        public OrtInferenceSession(ExecutionProviders provider = ExecutionProviders.CPU)
+        {
+            _sessionOptions = new SessionOptions();
+            switch (_executionProvider)
+            {
+                case ExecutionProviders.CPU:
+                    break;
+                case ExecutionProviders.NNAPI:
+                    _sessionOptions.AppendExecutionProvider_Nnapi();
+                    break;
+                case ExecutionProviders.CoreML:
+                    _sessionOptions.AppendExecutionProvider_CoreML();
+                    break;
+                case ExecutionProviders.XNNPACK:
+                    _sessionOptions.AppendExecutionProvider("XNNPACK");
+                    break;
+            }
+
+            // enable pre/post processing custom operators from onnxruntime-extensions
+            _sessionOptions.RegisterOrtExtensions();
+
+            _perfStats = new PerfStats();
+        }
+
+        // async task to create the inference session which is an expensive operation.
+        public async Task Create()
+        {
+            // create the InferenceSession. this is an expensive operation so only do this when necessary.
+            // the InferenceSession supports multiple calls to Run, including concurrent calls.
+            var modelBytes = await Utils.LoadResource("test_data/model.onnx");
+
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            _inferenceSession = new InferenceSession(modelBytes, _sessionOptions);
+            stopwatch.Stop();
+            _perfStats.LoadTime = stopwatch.Elapsed;
+
+            (_inputs, _expectedOutputs) = await Utils.LoadTestData();
+
+            // warmup
+            Run(1, true);
+        }
+
+        public void Run(int iterations = 1, bool isWarmup = false)
+        {
+            // do all setup outside of the timing
+            var runOptions = new RunOptions();
+            var outputNames = _inferenceSession.OutputNames;
+
+            _perfStats.ClearRunTimes();
+
+            // var stopwatch = new Stopwatch();
+
+            for (var i = 0; i < iterations; i++)
+            {
+                // stopwatch.Restart();
+                var stopwatch = new Stopwatch();
+                stopwatch.Start();
+
+                using IDisposableReadOnlyCollection<OrtValue> results =
+                    _inferenceSession.Run(runOptions, _inputs, outputNames);
+
+                stopwatch.Stop();
+
+                if (isWarmup)
+                {
+                    _perfStats.WarmupTime = stopwatch.Elapsed;
+
+                    // TODO: validate the output if there's expected output and validation is enabled
+                    // If we're doing a perf test we don't need to do validation. We could possibly do a warmup request
+                    // in Create and validate there so it's clearly a one-off.
```

Comment:
is there more to do for this comment? looks like it already does validation only on warmup